plugins/modules/route53.py

```diff
@@ -593,113 +513,106 @@ def main():
-    for v in value_in:
-        if alias_in:
-            wanted_rset.set_alias(alias_hosted_zone_id_in, v, alias_evaluate_target_health_in)
-        else:
-            wanted_rset.add_value(v)
-
-    need_to_sort_records = (type_in == 'CAA')
-
-    # Sort records for wanted_rset if necessary (keep original list)
-    unsorted_records = wanted_rset.resource_records
-    if need_to_sort_records:
-        wanted_rset.resource_records = sorted(unsorted_records)
-
-    sets = invoke_with_throttling_retries(conn.get_all_rrsets, zone_id, name=record_in,
-                                          type=type_in, identifier=identifier_in)
-    sets_iter = iter(sets)
-    while True:
-        try:
-            rset = invoke_with_throttling_retries(next, sets_iter)
-        except StopIteration:
-            break
-        # Need to save this changes in rset, because of comparing rset.to_xml() == wanted_rset.to_xml() in next block
-        rset.name = decode_name(rset.name)
-
-        if identifier_in is not None:
-            identifier_in = str(identifier_in)
-
-        if rset.type == type_in and rset.name.lower() == record_in.lower() and rset.identifier == identifier_in:
-            if need_to_sort_records:
-                # Sort records
-                rset.resource_records = sorted(rset.resource_records)
-            found_record = True
-            record = to_dict(rset, zone_in, zone_id)
-            if command_in == 'create' and rset.to_xml() == wanted_rset.to_xml():
-                module.exit_json(changed=False)
-
-        # We need to look only at the first rrset returned by the above call,
-        # so break here. The returned elements begin with the one matching our
-        # requested name, type, and identifier, if such an element exists,
-        # followed by all others that come after it in alphabetical order.
-        # Therefore, if the first set does not match, no subsequent set will
-        # match either.
-        break
+    aws_record = get_record(route53, zone_id, record_in, type_in, identifier_in)
+
+    resource_record_set = remove_none_values_from_dict({
+        'Name': record_in,
+        'Type': type_in,
+        'Weight': weight_in,
+        'Region': region_in,
+        'Failover': failover_in,
+        'TTL': ttl_in,
+        'ResourceRecords': [dict(Value=value) for value in value_in],
+        'HealthCheckId': health_check_in,
+    })
+
+    if alias_in:
+        resource_record_set['AliasTarget'] = dict(
+            HostedZoneId=alias_hosted_zone_id_in,
+            DNSName=value_in[0],
+            EvaluateTargetHealth=alias_evaluate_target_health_in
+        )
+
+    # On CAA records order doesn't matter
+    if type_in == 'CAA':
+        resource_record_set['ResourceRecords'] = sorted(resource_record_set['ResourceRecords'], key=itemgetter('Value'))
+
+    if command_in == 'create' and aws_record == resource_record_set:
+        module.exit_json(changed=False)
 
     if command_in == 'get':
```

Comment:
You mentioned being interested in helping with community.aws in general.  Not as part of this PR, but a separate PR we should probably deprecate the 'get' command here.

You'll notice that in tests/sanity/ignore-2.11.txt there's an entry 
```
plugins/modules/route53.py validate-modules:parameter-state-invalid-choice
```

https://docs.ansible.com/ansible/latest/dev_guide/developing_modules_best_practices.html#scoping-your-module-s
```
Do not add get, list or info state options to an existing module - create a new _info or _facts module
```

'get' commands are generally better handled by a dedicated _info module.  I believe route53_info can be used to do this so if you're willing to raise a follow up PR it would be appreciated